service/src/main/java/bio/terra/pipelines/app/configuration/internal/ImputationConfiguration.java

```diff
@@ -10,11 +11,43 @@
 @Getter
 @Setter
 public class ImputationConfiguration {
-  private Long cromwellSubmissionPollingIntervalInSeconds;
-  private List<String> inputKeysToPrependWithStorageUrl;
 
-  private String storageWorkspaceStorageUrl;
+  private Long cromwellSubmissionPollingIntervalInSeconds;
+  private List<String> inputKeysToPrependWithStorageWorkspaceContainerUrl;
+  private String storageWorkspaceContainerUrl;
+  private Map<String, String> inputsWithCustomValues;
   private boolean useCallCaching;
   private boolean deleteIntermediateFiles;
   private boolean useReferenceDisk;
+
+  public ImputationConfiguration(
+      Long cromwellSubmissionPollingIntervalInSeconds,
+      List<String> inputKeysToPrependWithStorageWorkspaceContainerUrl,
+      String storageWorkspaceContainerUrl,
+      Map<String, String> inputsWithCustomValues,
+      boolean useCallCaching,
+      boolean deleteIntermediateFiles,
+      boolean useReferenceDisk) {
+    this.cromwellSubmissionPollingIntervalInSeconds = cromwellSubmissionPollingIntervalInSeconds;
+    this.inputKeysToPrependWithStorageWorkspaceContainerUrl =
+        inputKeysToPrependWithStorageWorkspaceContainerUrl;
+
+    if (storageWorkspaceContainerUrl.endsWith("/"))
```

Comment:
any reason why we're checking this? are there other places this could mess with us that we havent accounted for?

Reply:
there shouldn't be other places where this matters, but now that we're constructing paths more smartly this shouldn't be necessary. happy to remove it.